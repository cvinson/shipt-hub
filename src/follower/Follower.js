import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Avatar from 'material-ui/Avatar';
import { ListItem } from 'material-ui/List';

const Follower = ({ avatar_url, login, style }) => {
  return (
    <Link to={`/${login}`}>
      <ListItem
        leftAvatar={<Avatar src={avatar_url} />}
        primaryText={login}
        style={style} />
    </Link>
  );
};

Follower.PropTypes = {
  avatar_url: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  style: PropTypes.node.isRequired,
};

export default Follower;
