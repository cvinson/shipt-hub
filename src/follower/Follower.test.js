import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Follower from './Follower';

describe('<Follower />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Follower
        avatar_url="https://avatars1.githubusercontent.com/u/590431?v=4"
        login="cvinson"
        style={{ color: 'green' }} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
