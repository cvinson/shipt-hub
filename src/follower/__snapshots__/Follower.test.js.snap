// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Follower /> renders correctly 1`] = `
<Link
  replace={false}
  to="/cvinson"
>
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    leftAvatar={
      <Avatar
        size={40}
        src="https://avatars1.githubusercontent.com/u/590431?v=4"
      />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="cvinson"
    primaryTogglesNestedList={false}
    secondaryTextLines={1}
    style={
      Object {
        "color": "green",
      }
    }
  />
</Link>
`;
